aexpect/remote_door.py

```diff
@@ -759,19 +759,29 @@ def share_remote_objects(session, control_path, host="localhost", port=9090,
     logging.getLogger("Pyro4").setLevel(10)
 
 
-def import_remote_exceptions(exceptions):
+def import_remote_exceptions(exceptions, module=None):
     """
     Make serializable all remote custom exceptions.
 
     :param exceptions: module dot accessible exception names
     :type exceptions: [str]
+    :param module: name of a module whose custom exceptions will first be
+                   detected and then automatically imported (optional)
+    :type exceptions: str or None
 
     .. note:: This wouldn't be needed if we were using the Pickle serializer but its
         security problems at the moment made us prefer the serpent serializer paying
         for it with some extra setup steps and this method.
     """
-    # ..todo:: need a more dynamic way of determining the exceptions like examining
-    # the errors modules and even extracting the exception classes from there
+    def list_module_exceptions(mod):
+        module = __import__(mod)
+        exceptions = []
+        for name in module.__dict__:
+            if (issubclass(module.__dict__[name], Exception) or name.endswith('Error')):
+                exceptions.append(name)
+        return exceptions
+    exceptions = list_module_exceptions(module) if module else exceptions
```

Comment:
I can't say I completely understand the `import_remote_exceptions` method, but this addition which just detects all exceptions makes sense. Would be nice to add a selftest for this. IIUC it should be possible even on a single host, right?

Reply:
Thanks for noticing, it is ideal to batch a unit test together with the newer more complex functionality here. Done in the push I just did.